src/Centreon/Domain/PlatformTopology/PlatformTopologyService.php

```diff
@@ -358,14 +366,14 @@ private function checkHostname(?string $stringToCheck): void
 
     /**
      * Check consistency of each mandatory and optional parameters
-     * @param Platform $platform
+     * @param PlatformInterface $platform
      * @throws EngineException
      * @throws EntityNotFoundException
      * @throws MonitoringServerException
      * @throws PlatformConflictException
      * @throws PlatformException
      */
-    private function checkEntityConsistency(Platform $platform): void
+    private function checkEntityConsistency(PlatformInterface $platform): void
```

Comment:
i think this method could be deleted and  all those checks handled by a Factory. Just an idea but it could alleviate the service.